### PR TITLE
Grub2 bootloader CPU mitigations

### PR DESCRIPTION
**Pull request by yuumasato**
#### Description:

- Add rules to check and remediate the following grub2 boot options
  - `l1tf`
  - `spec_store_bypass_disable`
  - `spectre_v2`
  - `mds`
  - `rng_core.default_quality`
- This also fixes a bug on the variable parametrization of `grub2_bootloader_argument` template.

### CODE REVIEW

File: linux_os/guide/system/bootloader-grub2/grub2_l1tf_argument/rule.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+documentation_complete: true
+
+title: 'Configure L1 Terminal Fault mitigations'
+
+description: |-
+    L1 Terminal Fault (L1TF) is a hardware vunerability which allows unprivileged
+    speculative access to data which is available in the Level 1 Data Cache when
+    the page table entry.
```
Comment (jan-cerny) [inline linux_os/guide/system/bootloader-grub2/grub2_l1tf_argument/rule.yml:8]:
```suggestion
    the page table entry isn't present.
```

File: linux_os/guide/system/bootloader-grub2/grub2_mds_argument/rule.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+description: |-
+    Microarchitectural Data Sampling (MDS) is a hardware vulnerability which allows unprivileged
+    speculative access to data which is available in various CPU internal buffers.
+
+    When performing store, load, L1 refill operations, processors write data into temporary
+    microarchitectural structures (buffers), and the data in the buffer can be forwarded to load
+    operations as an optimization.
+
+    Under certain conditions, data unrelated to the load memory address can be speculatively
```
Comment (jan-cerny) [inline linux_os/guide/system/bootloader-grub2/grub2_mds_argument/rule.yml:15]: I don't understand "load memory address" here. Is it target address of the "load" operation?

File: linux_os/guide/system/bootloader-grub2/grub2_spec_store_bypass_disable_argument/rule.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+documentation_complete: true
+
+title: 'Speculative Store Bypass Disable'
```
Comment (jan-cerny) [inline linux_os/guide/system/bootloader-grub2/grub2_spec_store_bypass_disable_argument/rule.yml:3]:
```suggestion
title: 'Disable Speculative Store Bypass'
```

File: linux_os/guide/system/bootloader-grub2/grub2_l1tf_argument/rule.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+documentation_complete: true
+
+title: 'Configure L1 Terminal Fault mitigations'
+
+description: |-
+    L1 Terminal Fault (L1TF) is a hardware vunerability which allows unprivileged
```
Comment (yuumasato) [inline linux_os/guide/system/bootloader-grub2/grub2_l1tf_argument/rule.yml:6]:
```suggestion
    L1 Terminal Fault (L1TF) is a hardware vulnerability which allows unprivileged
```

File: linux_os/guide/system/bootloader-grub2/grub2_mds_argument/rule.yml (hunk 2 above)
Comment (yuumasato) [reply linux_os/guide/system/bootloader-grub2/grub2_mds_argument/rule.yml:15]: Yes.

File: linux_os/guide/system/bootloader-grub2/grub2_spec_store_bypass_disable_argument/rule.yml (hunk 3 above)
Comment (yuumasato) [reply linux_os/guide/system/bootloader-grub2/grub2_spec_store_bypass_disable_argument/rule.yml:3]: How about the following title?
```suggestion
title: 'Configure Speculative Store Bypass Mitigation'
```

File: linux_os/guide/system/bootloader-grub2/grub2_mds_argument/rule.yml (hunk 2 above)
Comment (yuumasato) [reply linux_os/guide/system/bootloader-grub2/grub2_mds_argument/rule.yml:15]:
```suggestion
    Under certain conditions, data unrelated to the load operations can be speculatively
```

**Review by jan-cerny — APPROVED** [review]

**Review by jan-cerny — APPROVED** [review]
```
[jcerny@thinkpad scap-security-guide{pr/8325}]$ python3 tests/test_suite.py rule --libvirt qemu:///system ssgts_rhel8 grub2_l1tf_argument,grub2_mds_argument,grub2_spectre_v2_argument,grub2_rng_core_default_quality_argument,grub2_spec_store_bypass_disable_argument
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2022-03-22-0922/test_suite.log
WARNING - Script wrong_value_entries.fail.sh is not applicable on given platform
WARNING - Script invalid_rescue.pass.sh is not applicable on given platform
INFO - xccdf_org.ssgproject.content_rule_grub2_l1tf_argument
INFO - Script wrong_value.fail.sh using profile (all) OK
INFO - Script arg_not_there_etcdefaultgrub.fail.sh using profile (all) OK
INFO - Script arg_not_there_grubenv.fail.sh using profile (all) OK
INFO - Script correct_value.pass.sh using profile (all) OK
```

**Pull request merged**